src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1296,8 +1296,12 @@ public override BoundNode VisitEventAssignmentOperator(BoundEventAssignmentOpera
             return null;
         }
 
+        /// <summary>
+        /// Do not call for a local function.
+        /// </summary>
         protected virtual void VisitArguments(ImmutableArray<BoundExpression> arguments, ImmutableArray<RefKind> refKindsOpt, MethodSymbol method)
         {
+            Debug.Assert(!(method.OriginalDefinition is LocalFunctionSymbol));
```

Comment:
'method' can be null here, so consider using a condition like `method?.OriginalDefinition.MethodKind != MethodKind.LocalFunction`

Also, is there any situation where some constructed method's MethodKind differs from the OriginalDefinition's MethodKind? It feels like we may be able to just check `method?.MethodKind`?

Reply:
We check the OriginalDefinition here: https://github.com/dotnet/roslyn/pull/41899/files#diff-35d4070efa937f088ceaea2864b3279fL1165

I presume that's for a reason.